### PR TITLE
Add managed BO PREAD and BO PWRITE IOCTL

### PR DESCRIPTION
**Pull request by igorshr**
Add managed BO PREAD and BO PWRITE IOCTL
Return false on unmanaged BO PREAD and BO PWRITE

### CODE REVIEW

File: src/runtime_src/core/pcie/windows/shim.cpp [hunk 1]
```diff
@@ -1085,6 +1099,23 @@ xclUnmgdPread(xclDeviceHandle handle, unsigned int flags, void *buf, size_t coun
   return shim->unmgd_pread(flags, buf, count, offset);
 }
 
+ssize_t
+xclMgdPwrite(xclDeviceHandle handle, unsigned int flags, const void *buf, size_t count, uint64_t offset)
+{
```
Comment (sonals) [inline src/runtime_src/core/pcie/windows/shim.cpp]: size_t xclWriteBO(xclDeviceHandle handle, unsigned int boHandle, const void *src, size_t size, size_t seek)

File: src/runtime_src/core/pcie/windows/shim.cpp [hunk 2]
```diff
@@ -746,54 +746,68 @@ struct shim
...
+          xrt_core::message::
+              send(xrt_core::message::severity_level::XRT_ERROR, "XRT", "DeviceIoControl PWRITE failed with error %d", code);
+          return false;
+      }
+      return true;
   }
 
   ssize_t
-  unmgd_pread(unsigned int flags, void *buf, size_t size, uint64_t offset)
+      mgd_pread(unsigned int flags, void *buf, size_t size, uint64_t offset)
```
Comment (arpitxilinx) [inline src/runtime_src/core/pcie/windows/shim.cpp]: What is the use of passing in flags if it is not used?

File: src/runtime_src/core/pcie/windows/shim.cpp [hunk 3]
```diff
@@ -1085,6 +1099,23 @@ xclUnmgdPread(xclDeviceHandle handle, unsigned int flags, void *buf, size_t coun
...
+xclMgdPwrite(xclDeviceHandle handle, unsigned int flags, const void *buf, size_t count, uint64_t offset)
+{
+    xrt_core::message::
+        send(xrt_core::message::severity_level::XRT_DEBUG, "XRT", "xclMgdPwrite()");
+    auto shim = get_shim_object(handle);
+    return shim->mgd_pwrite(flags, buf, count, offset);
+}
+
+ssize_t
+xclMgdPread(xclDeviceHandle handle, unsigned int flags, void *buf, size_t count, uint64_t offset)
```
Comment (arpitxilinx) [inline src/runtime_src/core/pcie/windows/shim.cpp]: If we are changing the xclMgdPwrite to xclWriteBO, then we should change this to xclReadBO.

**Comment by sonals** [comment]
Feature merged via another PR

**Pull request closed without merging**